core/src/test/scala/org/broadinstitute/dsde/rawls/user/UserServiceSpec.scala

```diff
@@ -149,4 +151,81 @@ class UserServiceSpec extends FlatSpecLike with TestDriverComponent with Mockito
+      val petSAJson = "petJson"
+      runAndWait(rawlsBillingProjectQuery.create(project))
+
+      val mockSamDAO = mock[SamDAO]
+      when(mockSamDAO.userHasAction(SamResourceTypeNames.billingProject, project.projectName.value, SamBillingProjectActions.deleteBillingProject, userInfo)).thenReturn(Future.successful(true))
+      when(mockSamDAO.listAllResourceMemberIds(SamResourceTypeNames.billingProject, project.projectName.value, userInfo)).thenReturn(Future.successful(Set(userIdInfo)))
+      when(mockSamDAO.getPetServiceAccountKeyForUser(project.projectName.value, userInfo.userEmail)).thenReturn(Future.successful(petSAJson))
+      when(mockSamDAO.deleteUserPetServiceAccount(project.projectName.value, userInfo)).thenReturn(Future.successful())
+      when(mockSamDAO.deleteResource(SamResourceTypeNames.billingProject, project.projectName.value, userInfo)).thenReturn(Future.successful())
+
+      val mockGcsDAO = mock[GoogleServicesDAO]
+      when(mockGcsDAO.getUserInfoUsingJson(petSAJson)).thenReturn(Future.successful(userInfo))
+      when(mockGcsDAO.deleteProject(project.projectName)).thenReturn(Future.successful())
+
+      val userService = getUserService(dataSource, mockSamDAO, gcsDAO = mockGcsDAO)
+      val actual = userService.DeleteBillingProject(defaultBillingProjectName).futureValue
+
+      verify(mockGcsDAO).deleteProject(project.projectName)
+      actual shouldEqual RequestComplete(StatusCodes.NoContent)
+    }
+  }
+
+  // 403 when workspace exists in this billing project to be deleted
+  it should "fail with a 400 when workspace exists in this billing project to be deleted" in {
+    withEmptyTestDatabase { dataSource: SlickDataSource =>
+      val project = defaultBillingProject
+      // A workspace with the which namespaceName equals to defaultBillingProject's billing project name.
+      val workspace = Workspace(
+        defaultBillingProjectName.value,
+        testData.wsName.name,
+        UUID.randomUUID().toString,
+        "aBucket",
+        Some("workflow-collection"),
+        currentTime(),
+        currentTime(),
+        "test",
+        Map.empty
+      )
+
+      runAndWait(rawlsBillingProjectQuery.create(project))
+      runAndWait(workspaceQuery.save(workspace))
+
+      val mockSamDAO = mock[SamDAO]
+      when(mockSamDAO.userHasAction(SamResourceTypeNames.billingProject, project.projectName.value, SamBillingProjectActions.deleteBillingProject, userInfo)).thenReturn(Future.successful(true))
+
+      val userService = getUserService(dataSource, mockSamDAO)
+
+      val actual = intercept[RawlsExceptionWithErrorReport] {
+        Await.result(userService.DeleteBillingProject(defaultBillingProjectName), Duration.Inf).asInstanceOf[RequestComplete[ErrorReport]]
+      }
+      actual.errorReport.statusCode shouldEqual Option(StatusCodes.BadRequest)
+    }
+  }
+
+  // 404 when user doesn't have permission to delete billing project.
```

Comment:
remove this comment, it is wrong and the test name says it all anyway

Reply:
Done